calico_node/startup.go

```diff
@@ -51,22 +53,71 @@ func main() {
 	// Make sure we have a global cluster ID set.
 	log.Info("Ensuring a cluster guid is set")
 	ensureClusterGuid(c)
+
+	// Update the Node resource if necessary.
 }
 
 // ensureClusterGuid assigns a cluster GUID if one doesn't exist.
-func ensureClusterGuid(c api.Client) {
+func ensureClusterGuid(c bapi.Client) {
 	guid := hex.EncodeToString(uuid.NewV4().Bytes())
 	_, err := c.Create(&model.KVPair{
 		Key:   model.GlobalConfigKey{Name: "ClusterGUID"},
 		Value: guid,
 	})
 	if err != nil {
 		if _, ok := err.(errors.ErrorResourceAlreadyExists); !ok {
-			log.Warnf("Failed to set ClusterGUID: %s", err)
+			log.WithError(err).Warn("Failed to set ClusterGUID")
 			panic(err)
 		}
 		log.Infof("Using previously configured ClusterGUID")
 		return
 	}
 	log.Infof("Assigned ClusterGUID %s", guid)
 }
+
+// Update the current Node resource configuration (if it exists).
+func updateNodeResource() {
```

Comment:
@caseydavenport:  I realize that this will overlap a lot with that work -- so I am expecting some overlap.  Be aware this work is WIP and a little premature for review because these is little prototyping and also a little more investigation required.   At the moment I am not sure if interface querying is sufficient or whether we should be querying the AWS metadata service.  I needed internet access to be able to do that which I didn't have at the time.